Test/CaptureFileReaderDeviceTest.cs

```diff
@@ -125,19 +125,22 @@ public void SetFilter()
 
             RawCapture rawPacket;
             int count = 0;
+            CaptureEventArgs e;
+            int retval;
             do
             {
-                rawPacket = device.GetNextPacket();
-                if (rawPacket != null)
+                retval = device.GetNextPacket(out e);
+                if (retval == 0)
                 {
+                    rawPacket = e.Packet;
```

Comment:
I think we should revert those kind of changes to the tests, and add extension methods, so that 
* User that are used to the non Span API, don't have to do big refactoring like this in their code base.
* We can tell that tests are passing as is, without being modified.

Reply:
@kayoub5 let me see about putting that in place and reverting those changes to the tests. And by extension method you mean a GetNextPacket that converts to RawPacket so the API doesn’t change? Would we drop the extension method after a while? Imo we aren’t necessarily helping clean up the API if we are preserving so many ways to do things.

Reply:
> you mean a GetNextPacket that converts to RawPacket so the API doesn’t change?

Yes

> Would we drop the extension method after a while?

The idea is that we keep the same API available, but keep the classes that contain the logic as clean as possible.
If we at some point deprecate an API, we simply mark an extension method as `[Obsolete]`, then remove it at a later release, to give users time to migrate.

> we aren’t necessarily helping clean up the API if we are preserving so many ways to do things.

The Span API is more performant, but also more difficult to use due to its constraints, so having 2 ways (easy vs performant) to do things is not such a bad idea.

Reply:
I thought the same but it’s pretty easy to use the CaptureEventArgs version. So much so we should consider dropping the old versions now so people moving to 6.x can make the change. It’s a good time to minimize the API. We try not to do so in minor rev changes.